src/main/java/com/google/sps/servlets/RetrieveActivitiesServlet.java

```diff
@@ -0,0 +1,105 @@
+package com.google.sps.servlets;
+
+import java.io.IOException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import com.google.gson.Gson;
+
+import com.google.sps.data.Activity;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.ArrayList;
+
+/** 
+* Servlet that retrieves actvities of a specified category.
+*/
+@WebServlet("/retrieve-activities")
+public class RetrieveActivitiesServlet extends HttpServlet {
+  private static final String GAMES_CATEGORY = "games";
+  private static final String VIDEOS_CATEGORY = "videos";
+  private static final String ARTICLES_CATEGORY = "articles";
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String activityType = getActivityType(request);
+    List<Activity> activities;
+
+    switch(activityType) {
+      case GAMES_CATEGORY:
+        activities = getGames();
+        break;
+      case VIDEOS_CATEGORY:
+        activities = getVideos();
+        break;
+      case ARTICLES_CATEGORY:
+        activities = getArticles();
+        break;
+      default:
+        activities = null;
+        break;
+    }
+
+    // Convert activity list to json string
+    String activitiesJson = new Gson().toJson(activities);
+
+    response.setContentType("application/json;");
+    response.getWriter().println(activitiesJson);
+  }
+
+  /**
+  * Retrieves type of activity from request parameters.
+  */
+  private static String getActivityType(HttpServletRequest request) {
+    String name = "activity-type", defaultValue = "";
+    return getParameter(request, name, defaultValue);
+  }
+
+  private static List<Activity> getGames() {
+    List<Activity> games = Arrays.asList(
+      new Activity("Pictionary", GAMES_CATEGORY, "www.pictionary.com"),
```

Comment:
We should move all data to its own file. Changing the data should be outside of servlet code. Move it to either its own java file, text file, or database.

Reply:
Note: adding a TODO and doing this separation in another pull request is perfectly fine as well.

Reply:
Fixed it.